qa/zcash/performance-measurements.sh

```diff
@@ -406,6 +421,73 @@ case "$1" in
                 exit 1
         esac
         ;;
+    heaptrack)
```

Comment:
Rather than adding a new `heaptrack` option in parallel with the `memory` option, I think we could just update the `memory` option to replace Massif (as the only purpose of the `memory` option is to collect our memory benchmarks, so if we're moving to `heaptrack` for that due to performance improvements, then we don't need to retain Massif support).

Reply:
I wasn't sure if we wanted to preserve the old legacy targets. They do still work but are rather clunky compared to heaptrack

Reply:
I don't believe anyone other than us was using these targets, and I'm certainly not going to spin up Massif on a regular basis. So let's just migrate them.

Reply:
I will gladly remove the joys of massif lol. Just wanted to be sure. Ty